### DISCUSSION TITLE
osproxy mode - redirect outgoing http to https

### DISCUSSION THREAD

**Discussion by BH4rd** [mitmproxy/mitmproxy#6341] [Q&A] [answered]
Hi,

first of all thank you very much to the devs for investing so much effort in this extremely powerful software!

I have a question about the osproxy mode. I'm running Windows Server 2012 R2 on a machine (can't upgrade to newer version) and I'd like to redirect outgoing http traffic to a specific server still sending it to the original server but via https and different port, so I start mitmproxy with the following command:
`mitmweb --mode osproxy:msedge.exe --showhost --ssl-insecure -s C:\temp\redirect.py`

msedge.exe is just for testing purposes.

I tried with two different approaches in redirect.py.
Approach 1
```
from mitmproxy import http

def request(flow):
    if flow.request.port == 80:
        flow.request.port = 443
        flow.request.scheme = "https"
```

Approach 2
```
from mitmproxy import http

def request(flow):
    if flow.request.port == 80:
        flow.response = http.Response.make(302, b'',
                            http.Headers(Location='https://10.0.0.1:443/',
                                    Content_Length='0'))
```

Both resulting in two outgoing connections, one for http and one for https:
```
[13:31:59.281][192.168.0.1:60005] server connect 10.0.0.1:80
[13:31:59.307][192.168.0.1:60007] client connect
[13:31:59.307][192.168.0.1:60008] client connect
[13:31:59.315][192.168.0.1:60005] server connect 10.0.0.1:443
```

At the end of the day I want to block port 80 on our external network firewall, I simulated this with a connection attempt to port 81 where no service is running on the server, which should result in the same behaviour I guess. In this case I don't get a flow in mitmproxy at all.

I am quite new to mitmproxy and not an expert in network and web technology in general, can somebody please tell me where I am doing wrong? Or do you need further information?

I'd prefer something like approach 1, because I'm not sure, if the application which should be redirected even handles the 302 code.

**Comment by mhils** [comment, marked as answer]
Have you tried setting the https://docs.mitmproxy.org/stable/concepts-options/#connection_strategy option to lazy? :)

**Reply by BH4rd** [reply]
Yay, seems like this one did the trick, thank you so much! (feeling like a total newbie since I was not even aware of different connection strategies, never came across this one in all the Google results I searched through...)
